src/error.rs

```diff
@@ -135,6 +135,62 @@ pub enum ConvertError<A, S, V> {
     Validity(V),
 }
 
+impl<Src, Dst: ?Sized + Unaligned, S, V> From<ConvertError<AlignmentError<Src, Dst>, S, V>>
+    for ConvertError<Infallible, S, V>
+{
+    /// Infallibly discards the alignment error from this `ConvertError` since
+    /// `Dst` is unaligned.
+    ///
+    /// Since [`Dst: Unaligned`], it is impossible to encounter an alignment
+    /// error. This method permits discarding that alignment error infallibly
+    /// and replacing it with [`Infallible`].
+    ///
+    /// [`Dst: Unaligned`]: crate::Unaligned
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use core::convert::Infallible;
+    /// use zerocopy::*;
+    /// # use zerocopy_derive::*;
+    ///
+    /// // The only valid value of this type is the byte `0xC0`
+    /// #[derive(TryFromBytes, KnownLayout, Immutable)]
+    /// #[repr(u8)]
+    /// enum C0 { xC0 = 0xC0 }
+    ///
+    /// // The only valid value of this type is the byte sequence `0xC0C0`.
+    /// #[derive(TryFromBytes, KnownLayout, Immutable)]
+    /// #[repr(C)]
+    /// struct C0C0(C0, C0);
+    ///
+    /// #[derive(TryFromBytes, KnownLayout, Unaligned, Immutable)]
+    /// #[repr(C)]
+    /// struct Packet {
+    ///     magic_number: Unalign<C0C0>,
+    ///     mug_size: u8,
+    ///     temperature: u8,
+    ///     marshmallows: [[u8; 2]],
+    /// }
+    ///
+    /// impl Packet {
+    ///     pub fn parse(bytes: &[u8]) -> Result<&Packet, ConvertError<Infallible, SizeError<&[u8], Packet>, ValidityError<&[u8], Packet>>> {
```

Comment:
This type signature is awful. Should we figure out a way to express this using type aliases instead? Perhaps add `UnalignedConvertError` and `UnalignedCastError` type aliases?

Reply:
We have an alias for this: https://docs.rs/zerocopy/0.8.0-alpha.22/zerocopy/error/type.TryReadError.html

Reply:
Discussed offline; added an `UnalignedTryCastError`.